File: beneath-management/src/lib/redis.ts
import redis from "redis";
import logger from "./logger";

const url = process.env.REDIS_URL;
let client = null;
if (url) {
  client = redis.createClient(url);
  client.on("ready", () => {
    logger.info("Redis is running");
  });
  client.on("error", (err: any) => {
    logger.error("Redis error", err);
  });
} else {
  logger.info(`Running without Redis`);
}

export default client;

File: beneath-gateway/management/redis.go
package management

import (
	"log"

	"github.com/go-redis/redis"
)

func newRedis(redisURL string) *redis.Client {
	opts, err := redis.ParseURL(redisURL)
	if err != nil {
		log.Fatalf("redis: %s", err.Error())
	}

	client := redis.NewClient(opts)
	return client
}
